pkg/types/types.go

```diff
@@ -154,35 +154,78 @@ func (m *ComMatrix) WriteMatrixToFileByType(utilsHelpers utils.UtilsInterface, f
 	return nil
 }
 
-// GenerateMatrixDiff generates the diff between mat1 to mat2.
-func (m *ComMatrix) GenerateMatrixDiff(mat2 *ComMatrix) (string, error) {
-	colNames, err := getComMatrixHeadersByFormat(FormatCSV)
-	if err != nil {
-		return "", fmt.Errorf("error getting commatrix CSV tags: %v", err)
+// Combine generates a new sorted matrix from m and other with no duplications.
+func (m *ComMatrix) combine(other *ComMatrix) *ComMatrix {
+	// initial combined matrix with m.Matrix
+	combinedComDetails := m.Matrix
+
+	for _, cd := range other.Matrix {
+		// avoid duplications of comdetails on combined matrix
+		if !m.Contains(cd) {
+			combinedComDetails = append(combinedComDetails, cd)
+		}
 	}
 
-	diff := colNames + "\n"
+	// sort combined matrix
+	combinedComMatrix := &ComMatrix{Matrix: combinedComDetails}
+	combinedComMatrix.sort()
+	return combinedComMatrix
+}
+
+// markDiffBetweenMatrices map the cd's string of the matrices to ints in the following way:
+// cd which m contains but other doesn't --> 1
+// cd both m and other contains --> 0
+// cd which other doesn't contain but m does --> -1.
+func (m *ComMatrix) markDiffBetweenMatrices(other *ComMatrix) map[string]int {
+	mapComDetailToSign := make(map[string]int)
 
 	for _, cd := range m.Matrix {
-		if mat2.Contains(cd) {
-			diff += fmt.Sprintf("%s\n", cd)
-			continue
+		if other.Contains(cd) {
+			mapComDetailToSign[cd.String()] = 0
+		} else {
+			// m contains cd but other doesn't
+			mapComDetailToSign[cd.String()] = 1
 		}
-
-		// add "+" before cd's mat1 contains but mat2 doesn't
-		diff += fmt.Sprintf("+ %s\n", cd)
 	}
 
-	for _, cd := range mat2.Matrix {
+	for _, cd := range other.Matrix {
 		// Skip "rpc.statd" ports, these are randomly open ports on the node,
 		// no need to mention them in the matrix diff
 		if cd.Service == "rpc.statd" {
```

Comment:
I'm not sure if this is the right place to make this check

Reply:
As we discussed, the ss filter should be only in the origin, or am I wrong? 
Maybe we can address this subject in a separate PR?